File: Jongryul/01/10211.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");
const t = Number(input[0]);

let idx = 1;
const rlt = [];
for (let i = 0; i < t; i++) {
  const n = Number(input[idx++]);
  const dp = input[idx++].split(" ").map(Number);
  for (let j = 1; j < n; j++) {
    dp[j] = Math.max(dp[j], dp[j - 1] + dp[j]);
  }
  rlt.push(Math.max(...dp));
}
console.log(rlt.join("\n"));

File: Jongryul/01/10211.py
import sys
input = sys.stdin.readline

t = int(input())
for _ in range(t):
	n  = int(input())
	arr = list(map(int, input().split()))
	dp = [i for i in arr]
	for i in range(1, n):
		dp[i] = max(dp[i], dp[i-1] + dp[i])
	print(max(dp))